### PR TITLE
Add +0 to Array.prototype.indexOf to convert negative zero to positive zero

### PR DESCRIPTION
**Pull request by anba**
Fixes https://bugs.ecmascript.org/show_bug.cgi?id=4545

### CODE REVIEW

**Comment by bterlson** [comment]
Chakra seems to do this, but no other impls I can test convert -0 to +0.

```js
print("indexOf returns +0: " + (1/[true].indexOf(true, -0) > 0));
print("lastIndexOf returns +0: " + (1/[true].lastIndexOf(true, -0) > 0));
```

#### chakra
indexOf returns +0: true
lastIndexOf returns +0: true

#### spidermonkey
indexOf returns +0: false
lastIndexOf returns +0: false

#### d8
indexOf returns +0: false
lastIndexOf returns +0: false

#### node
indexOf returns +0: false
lastIndexOf returns +0: false

**Comment by anba** [comment]
This is the fix discussed last October in https://github.com/tc39/test262/issues/435#issuecomment-146688936. 

JSC and Nashorn also return +0.

File: spec.html [hunk 1]
```diff
@@ -29809,7 +29809,8 @@
           1. Let _n_ be ? ToInteger(_fromIndex_). (If _fromIndex_ is *undefined*, this step produces the value 0.)
           1. If _n_ &ge; _len_, return -1.
           1. If _n_ &ge; 0, then
-            1. Let _k_ be _n_.
+            1. Let _k_ be _n_ + (*+0*).
+            1. NOTE: Adding a positive zero converts *-0* to *+0*.
```
Comment (ljharb) [inline spec.html]: rather than a note, would it make more sense to just have "if n is -0, let k be +0, else let k be n"? Adding +0 to -0 to get +0 seems like an irrelevant implementation detail.

File: spec.html [hunk 2]
```diff
@@ -29879,7 +29880,9 @@
           1. Let _len_ be ? ToLength(? Get(_O_, `"length"`)).
           1. If _len_ is 0, return -1.
           1. If argument _fromIndex_ was passed, let _n_ be ? ToInteger(_fromIndex_); else let _n_ be _len_-1.
-          1. If _n_ &ge; 0, let _k_ be min(_n_, _len_ - 1).
+          1. If _n_ &ge; 0, then
+            1. Let _k_ be min(_n_, _len_ - 1).
+            1. NOTE: Applying the mathematical function min(&hellip;) converts *-0* to *+0*.
```
Comment (ljharb) [inline spec.html]: similar question here - perhaps an explicit step to handle `-0` rather than a note?

File: spec.html (hunk 1 above)
Comment (domenic) [inline spec.html]: I strongly agree!

Comment (anba) [inline spec.html]: Okay. I'll also update [20.3.1.15 TimeClip](http://tc39.github.io/ecma262/#sec-timeclip) to use an extra step.

**Comment by anba** [comment]
Updated the PR per the review comments, negative zero is now handled with an explicit step.

**Comment by ljharb** [comment]
thanks, much clearer imo, LGTM!
